src/frontend/ocamlmerlin/ocamlmerlin_server.ml

```diff
@@ -19,7 +19,12 @@ module Server = struct
     Os_ipc.context_setup context;
     let close_with return_code =
       flush_all ();
-      Os_ipc.context_close context ~return_code
+      Os_ipc.context_close context ~return_code;
+      (* The first flush may have failed due to IO errors, so flush again now that we've
+         reset our file descriptors.  This may cause us to send random output to the
+         server's stdout and stderr, but it's better than sending random output to the
+         next client to connect. *)
```

Comment:
Having confusing data in log files might do more harm than what is claimed here. If I saw malformed-looking data in a log file, I would think something seriously wrong is going on and spend a lot of time investigating. Does the data actually end up in a log file? I wonder if we should log (and flush) some extra text to the fd, like "Finishing flushing stale data. In rare cases, this message will be preceded by some data that was intended to be sent to a client that recently crashed."

Reply:
I think that's a good point and worth doing

Reply:
BTW, stdout/stderr are /dev/null for ocamlmerlin-server on my box, which is probably the case everywhere, so it's unlikely to actually matter.